dmm.ts

```diff
@@ -142,48 +80,105 @@ async function getDenoLandDatabase (): Promise<any> {
 
 /**
  * @description
- * Grabs deno's database.json file and checks all the module names against their git repository
+ * Constructs the object representations for the users modules, that contains all the information about those modules,
+ * needed to: run any queries on them, or log information about them.
+ *     1. Reads `deps.ts` and turns each dependency into a module object
+ *     2. Adds a the github url to each object using Deno's database.json and the modules name
+ *     3. Adds the latest version to each object using the github url
  *
- * @param {Module[]} modules
+ * @param {string[]} modulesForPurpose Specific modules instead of checking all. Leave empty if all
+ * @param {string} purpose The purpose. Purely for logging purposes, eg "check" or "update"
  *
- * @return {Module[]} The same passed in parameter but with a new `repo` property
+ * @return {Module[]} An array of objects, with each object containing information about each module
  */
-export async function addGitHubUrlForModules (modules: Module[]): Promise<Module[]> {
-    console.info('Fetching GitHub urls...')
+async function constructModulesDataFromDeps (modulesForPurpose: string[], purpose: string): Promise<Module[]|boolean> {
+    async function getLatestReleaseOfGitHubRepo (githubURL: string): Promise<string> {
+        const res = await fetch(githubURL + "/releases/latest");
+        const splitUrl: string[] = res.url.split('/');
+        const latestVersion: string = splitUrl[splitUrl.length - 1];
+        return latestVersion
+    }
+
     const denoDatabase = await getDenoLandDatabase()
-    modules.forEach(module => {
-        if (module.std === false) {
-            // 3rd party
-            module.repo = "https://github.com/" + denoDatabase[module.name].owner + "/" + denoDatabase[module.name].repo
-        } else {
-            // std
-            module.repo = "https://github.com/denoland/deno/std/" + module.name
+
+    // Solely read the users `deps.ts` file
+    console.info('Reading deps.ts to gather your dependencies...')
+    const usersWorkingDir: string = "."
+    const depsContent: string = decoder.decode(Deno.readFileSync(usersWorkingDir + "/deps.ts")); // no need for a try/catch. The user needs a deps.ts file
+    // Turn lines that import from a url into a nice array
+    const listOfDeps: string[] = depsContent.split('\n').filter(line =>
+        line.indexOf('https://deno.land') !== -1
+    )
+    // Collate data for each module imported
+    let modules: Array<Module> = [];
+    for (const dep of listOfDeps) {
+        // Get if is std, imported version, name and deno.land url from the import line
+        const std: boolean = dep.indexOf("https://deno.land/std") >= 0
+        const denoLandURL: string = dep.substring(
+            dep.lastIndexOf("from ") + 5,
```

Comment:
the `+ 5` here kind of seems flimsy. i feel like there'd be an edge case where `+ 5` wouldn't give you what you expect

Reply:
Yeah you're right, ill try make it a bit more cleaner now

Reply:
May as well get it out of the way now tbh